### DISCUSSION TITLE
是否有办法保留第一轮延迟测试筛选后的ip

### DISCUSSION THREAD

**Discussion by CrayonSSR** [XIU2/CloudflareSpeedTest#470] [提问] [answered]
![image](https://github.com/XIU2/CloudflareSpeedTest/assets/93774795/7f30af22-22c6-467c-9911-e252b653ffc1)

从5956个ip测试延时，筛选出可用的ip，是一件比较费时的事情。

尤其是，最终我所需要的ip仅仅只有10-20个。

我是否有办法，直接保存延迟测速后筛选出的1000多个ip，下次只需从这1000个开始筛选。

相比之下，最多也只需要额外每隔几个月从5956个ip再筛选一次。

**Comment by XIU2** [comment]
很简单，写个脚本：

> `ip.txt` 文件是所有 Cloudflare IP 段数据，用于完整测速  
> `result.csv` 文件是每次测速完成后输出的完整结果信息  
> `ip1.txt` 文件是每次测速完成后根据 `result.csv` 整理的 IP 数据，用于下次测速

先判断 `ip1.txt` 文件是否存在，如果存在则运行 CloudflareST 时指定 `-f ip1.txt`，如果不存在则指定 `-f ip.txt`
运行测速完成后，读取结果文件 `result.csv` 内容，排除第一行表头，然后取第一列的 IP 信息写入到 `ip1.txt` 文件内。


> 注意，上面我说的是每次测速都是从上次的测速结果中测速挑选，然后再将本次的结果写入 `ip1.txt` 文件用于下次测速，这样可能每测速一次，结果就会少一些，越来越少。  
> 如果你是想要后续的每一次测速都只是单纯的从第一次的结果中（即那 1000 个 IP）测速，那么上面的需要改改：  
> 可以改成如果 `ip1.txt` 文件存在，那么就只是单纯的指定该文件测速，而不再执行后续的提取 IP 信息写入 `ip1.txt` 的步骤了。

****

后续再运行该脚本时，因为 `ip1.txt` 文件存在，因此会测速该文件内（也就是上次测速的结果）的 IP 地址了。

而当你想要 "每隔几个月从5956个ip再筛选一次" 时，只需要删除 `ip1.txt` 即可，这样你下次运行脚本就会因为缺少该文件而改用 `ip.txt` 里的 IP 段数据来测速了。

****

你是要在 Linux 系统下运行（Shell 脚本）？还是在 Windows 系统下运行（Bat 脚本）？你可以找 ChatGPT 写，也可以找我写。

但需要注意的是，使用默认的 `ip.txt` 测速时，是对所有 IP 段的最后一段（/24）随机取一个 IP，因此其实是在 1518780 个 IP 中随机抽取了 5956 个做测速，而如果你对其结果的 1000 个 IP 再去测速，虽然因为数量少了而减少了测速时间，但结果质量可能反而更差了。

最好的方法还是每次测速时注意一下优质 IP 都属于哪些 IP 段的，最后挑选出一些容易出现优质 IP 的 IP 段，然后后续就只测速这些优质 IP 段（相对于你来说），这样即保证了质量，也减少了测速时间。

**Reply by CrayonSSR** [reply]
您可能误会我的意思了。

我想要保留的**延迟测速**的结果，而非**延迟测试+下载测速**后的结果。

我标题的第一轮指的是，当我启动程序的时候 CloudflareST.exe 的输出日志，会显示两轮测试，分别是延迟测试和下载测速。

![image](https://github.com/XIU2/CloudflareSpeedTest/assets/93774795/9e83539f-70af-4892-b442-e7dc1e2566d8)

**Reply by XIU2** [reply, marked as answer]
那你加上 `-dd` 参数不就行了？
